GitHubActionsTestLogger/CliOptionsProvider.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.Testing.Platform.CommandLine;
+using Microsoft.Testing.Platform.Extensions;
+using Microsoft.Testing.Platform.Extensions.CommandLine;
+
+namespace GitHubActionsTestLogger;
+
+internal sealed class CliOptionsProvider(GitHubTestReporterExtension extension) : ICommandLineOptionsProvider
+{
+    public const string ReportGitHubOption = "report-github";
```

Comment:
Maybe instead of `const string`, we could have an initialized `static readonly CommandLineOption` here, so it's easier to refer to it later, for example in `ValidateOptionArgumentsAsync` (wouldn't need to compare names then).